### PR TITLE
Bugfix/artesca 14165 capacity decimals issue

### PR DESCRIPTION
**Pull request by JeanMarcMilletScality**
Issue with JS floating number precision

### CODE REVIEW

File: src/react/ui-elements/Veeam/VeeamCapacityModal.tsx [hunk 1]
```diff
@@ -29,10 +29,14 @@ const schema = Joi.object({
     .min(1)
     .max(1024)
     .custom((value, helpers) => {
-      if (!Number.isInteger(value * 100)) {
-        return helpers.message({
-          custom: '"capacity" must have at most 2 decimals',
-        });
+      const stringValue = value.toString();
+      if (stringValue.includes('.')) {
```
Comment (MonPote) [inline src/react/ui-elements/Veeam/VeeamCapacityModal.tsx]: I think you should add a test for that

File: src/react/ui-elements/Veeam/VeeamCapacityModal.test.tsx [hunk 2]
```diff
@@ -52,6 +53,16 @@ describe('VeeamCapacityModal', () => {
...
 
+  it('should enabled edit button when value is valid', async () => {
+    await userEvent.click(selectors.editBtn());
+
+    await userEvent.clear(selectors.capacityInput());
+    expect(selectors.editModalBtn()).toBeDisabled();
+    await userEvent.type(selectors.capacityInput(), '2.2');
+
+    expect(selectors.editModalBtn()).toBeEnabled();
+  });
```
Comment (JBWatenbergScality) [inline src/react/ui-elements/Veeam/VeeamCapacityModal.test.tsx:64]: nit: aren't we missing a test to check that numbers with more than 2 decimals are invalid ?

**Review by JBWatenbergScality — APPROVED** [review]

File: src/react/ui-elements/Veeam/VeeamCapacityModal.test.tsx (hunk 2 above)
Comment (JeanMarcMilletScality) [reply src/react/ui-elements/Veeam/VeeamCapacityModal.test.tsx:64]: There is a test for this:
`  it('should validate capacity value correctly : number with more than 2 decimals', async () => `

**Pull request merged**